peachjam/middleware.py

```diff
@@ -48,6 +49,25 @@ def process_request(self, request):
             del request.META["HTTP_ACCEPT_LANGUAGE"]
 
 
+class SetPreferredLanguageMiddleware(MiddlewareMixin):
+    def process_response(self, request, response):
+        if hasattr(request, "set_language") or hasattr(response, "set_language"):
+            lang_code = getattr(request, "set_language", None) or getattr(
+                response, "set_language", None
+            )
```

Comment:
i wish this was cleaner but here we have to check both request and response. the signal only has the request and the form valid method only has the response :smiling_face_with_tear:

Reply:
changed this to only use request. it works okay